app/src/main/java/com/stl/skipthelibrary/Activities/MapBoxActivity.java

```diff
@@ -95,14 +96,16 @@ public void onClick(View view) {
                 Gson gson = new Gson();
                 Intent intent=new Intent();
                 intent.putExtra("Location", gson.toJson(location));
+                intent.putExtra("username",getIntent().getExtras().getString("username"));
```

Comment:
@tymoorej @BrynL-F This here is a little funky friends, wanna instead get the extras in on create and have a activity variable for username? Idk what you think but this looks weird to me